runtime/cudaq/cudaq.cpp

```diff
@@ -271,6 +274,17 @@ void unset_noise() {
 }
 
 void set_random_seed(std::size_t seed) { nvqir::setRandomSeed(seed); }
+
+int num_available_gpus() {
+#ifdef CUDAQ_HAS_CUDA
+  int nDevices;
+  cudaGetDeviceCount(&nDevices);
```

Comment:
Perhaps we should check/handle the returned error code of this call (e.g., could return 0 as if no device is present)?

Reply:
Yes, I think that would make sense. What errors would you suggest to catch and ignore? 
I think when no gpu is available, this will simply return 0. That is exactly the situation for the python wheels, that get built with CUDAQ_HAS_CUDA set, but no gpu in the environment that runs the tests (and the tests call num_available_gpus to determine if they should be skipped). Am I overlooking something?

Reply:
Initializing it to 0 makes sense (updated), but I am inclined to leave the error regarding any incompatible driver versions visible; this is something that a user should probably know about.